### PR TITLE
Make toolchain detection more resilient

### PR DESCRIPTION
**Pull request by ljacomet**
Failing to run `java_home` on MacOS will simply return no installations
instead of failing the build.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RFM

File: subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java [hunk 1]
```diff
@@ -44,9 +49,17 @@ public OsXInstallationSupplier(ExecHandleFactory execHandleFactory, ProviderFact
...
-            final Reader output = executeJavaHome();
-            final Set<File> javaHomes = new OsXJavaHomeOutputParser().parse(output);
-            return javaHomes.stream().map(this::asInstallation).collect(Collectors.toSet());
+            try {
+                final Reader output = executeJavaHome();
+                final Set<File> javaHomes = new OsXJavaHomeOutputParser().parse(output);
+                return javaHomes.stream().map(this::asInstallation).collect(Collectors.toSet());
+            } catch (ExecException e) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.info("Java Toolchain auto-detection failed for MacOS system JVMs", e);
```
Comment (melix) [inline subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java:58]: :o: Why not directly `LOGGER.debug`?

**Review by melix — CHANGES_REQUESTED** [review]

File: subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java (hunk 1 above)
Comment (ljacomet) [reply subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java:58]: My take is that at info level you know that MacOS system JVMs were not detected, which feels like the right level of information.
In debug you know the root cause of that problem.

Comment (melix) [reply subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java:58]: Ok I see. Can I suggest to use a less dramatic version for `info`?

> Java toolchain auto-detection failed to find a local installation

Comment (ljacomet) [reply subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java:58]: Will do that, once CI gives me the green light.

**Review by melix — APPROVED** [review]

File: subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java (hunk 1 above)
Comment (ljacomet) [reply subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/OsXInstallationSupplier.java:58]: 🤦 Ok, there was an issue indeed. That specific `LOGGER.info` should have been `LOGGER.debug` indeed. Sorry, totally misinterpreted your comment.

Will use the following text:

> "Java Toolchain auto-detection failed to find local MacOS system JVMs"

**Pull request merged**